### DISCUSSION TITLE
using jwt

### DISCUSSION THREAD

**Discussion by Carmel558** [adonisjs/core#4400] [General]
I wanted to use the jwt token for authentication.
By following the documentation. I encounter the following error.

" SyntaxError: The requested module '@adonisjs/auth' does not provide an export named 'providers'  "

Please ask for your help

**Comment by RomainLanz** [comment]
Hey @Carmel558! 👋🏻 

This is an error in our documentation, it should be updated soon.

**Reply by Carmel558** [reply]
Ok thank you
For the moment, there are tips for integrating jwt into my adonis project ?

And also I want to use authentication in my project. I want to generate a token upon login and with it the user can interact with the api.
What do you suggest as an authentication method?

**Reply by RomainLanz** [reply]
> For the moment, there are tips for integrating jwt into my adonis project ?

Wait for the updated doc.

> What do you suggest as an authentication method?

If your API and your application are on the same domain. I would go with the [session guard](https://docs.adonisjs.com/guides/auth-session-guard).

If your API is on another domain than your application, or you are building a mobile/native application, I would go with the [access token guard](https://docs.adonisjs.com/guides/auth-access-tokens-guard).

Nothing stops you from having both if you match both cases.

JWT are needed only for very specific needs.

**Reply by Carmel558** [reply]
Thank you very much, but access-tokens-guard does not offer to refresh the token.
I need to keep the user logged in by refreshing the token to allow them to keep the connection even when the token expires

**Reply by thetutlage** [reply]
Do you really need a refresh token with an opaque access token?

**Reply by Carmel558** [reply]
Yes, I need it for the project
